src/main/java/duke/UI.java

```diff
@@ -0,0 +1,147 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.task.Task;
+
+/**
+ * UI class which handles all the interactions with the user.
+ */
+public class UI {
```

Comment:
```suggestion
public class Ui {
```

Should your class name be in lowercase instead? The [Naming](https://se-education.org/guides/conventions/java/intermediate.html#naming) conventions of the module coding standard state that "abbreviations and acronyms should not be uppercase when used as a part of a name".